pkg/controller/controller_utils.go

```diff
@@ -124,6 +127,20 @@ func GetOwnerRef(tc *v1alpha1.TidbCluster) metav1.OwnerReference {
 	}
 }
 
+// GetInitOwnerRef returns TiDBInitializer's OwnerReference
+func GetInitOwnerRef(ti *v1alpha1.TidbInitializer) metav1.OwnerReference {
+	controller := true
+	blockOwnerDeletion := true
+	return metav1.OwnerReference{
+		APIVersion:         InitControllerKind.GroupVersion().String(),
+		Kind:               InitControllerKind.Kind,
+		Name:               ti.GetName(),
+		UID:                ti.GetUID(),
+		Controller:         &controller,
+		BlockOwnerDeletion: &blockOwnerDeletion,
+	}
+}
```

Comment:
It is recommended to use `GenericControl.CreateOrUpdateXXX()` method, which will set owner reference automatically and make the reconcile process be idempotent.

Reply:
ok.